pywr/parameters/_parameters.pyx

```diff
@@ -670,6 +670,92 @@ cdef class ScenarioMonthlyProfileParameter(Parameter):
 
 ScenarioMonthlyProfileParameter.register()
 
+cdef class ScenarioWeeklyProfileParameter(Parameter):
+    """
+    This parameter provides a repeating annual profile with a weekly resolution. A
+    different profile is returned for each member of a given scenario
+    
+    Parameters
+    ----------
+    scenario: Scenario
+        Scenario object over which different profiles should be provided.
+    values : iterable, array
+        length of 1st dimension should equal the number of members in the scenario object 
+        and the length of the second dimension should be 52
+
+    """
+    def __init__(self, model, Scenario scenario, values, *args, **kwargs):
+        super().__init__(model, *args, **kwargs)
+        if values.ndim != 2:
+            raise ValueError("Factors must be two dimensional.")
+        if scenario._size != values.shape[0]:
+            raise ValueError("First dimension of factors must be the same size as scenario.")
+        if values.shape[1] != 52:
+            raise ValueError("52 values must be given for a weekly profile.")
+        self._values = values
+        self._scenario = scenario
+
+    cpdef setup(self):
+        super(ScenarioWeeklyProfileParameter, self).setup()
```

Comment:
Remove arguments to `super`

Reply:
I tried before but got this error:

`RuntimeError: super(): __class__ cell not found`

Looks like it is caused by the problem identified in this open cython issue - https://github.com/cython/cython/issues/2912 

As `__class__` in `setup` does not refer to the object class it is not possible to use `super` with no args.

What should we do here? It does look a bit inconsistent using args for `super` in `setup` when they are not used in `__init__`

Reply:
Yeah, OK. I forgot I was reading Cython!